### ISSUE TITLE
Refresh coin overview when new block detected

### ISSUE THREAD

**Issue by acityinohio** [blockcypher/explorer#178]
Got some feedback from a user that they'd love to see page refreshes when a new block is detected on the coin overview page (e.g. https://live.blockcypher.com/btc/) , thought it would be nice to track that issue here.

**Comment by acityinohio** [comment]
Oh, actually, correction: the transaction data page (confirmation count): e.g., https://live.blockcypher.com/btc/tx/791ee31890e457f687d0534bd11d0ac411092bc948ebeb48f68739ba8ed3b2e6/

**Comment by mflaxman** [comment]
The first comment about blocks makes sense (the explorer doesn't do this), but I'm confused re transactions since the explorer already does this? See here for example (you'll have to wait a bit for it to get a confirmation):
https://live.blockcypher.com/ltc/latest-unconfirmed-tx/

**Comment by acityinohio** [comment]
Oh my (double) bad again...so this works for a single conformation, but doesn't work after that? E.g., you have to reload for 2/3/4 confirmations (just tested this with that tx you linked to)...actually, looking through the code, I think it might be a WebSockets confirmation issue (e.g., the WebSocket stops sending events after 1 confirmation, which is currently a known issue we're working on).

...but yeah, forgetting the tx-confirmation stuff for the moment (as that's on the API side) the new-block reload would be nice. :)

**Comment by mflaxman** [comment]
It should work for multiple confirmations, but I just tested and it appeared not to. This is confusing, here's what's passed into the websocket (and I believe it worked in the past) that works for the 0->1 confirm just fine:
https://github.com/blockcypher/explorer/blob/39593ded91127201021496a284865269e7f2e446/templates/transaction_overview.html#L404-L416

You'll notice there's an ugly hack where it passes in both the tx-hash and the address, as I believe the address is required (despite that there is a tx-hash).

Here's what it looks like:
```
{
	'hash': "195ebc5d6ecbfb7f75a6fbab239bd9ba48e9d35c38942de4531e5a65aebc3669",
	'event': 'tx-confirmation',
	'address': "CFr99841LyMkyX5ZTGepY58rjXJhyNGXHf",
	'token': '31c49f33f35c85a8f4d9845a754f7c8e',
}
```

Here's how the confirmations param is explained in [the docs](http://dev.blockcypher.com/#event):
> optional Used in concert with the tx-confirmation event type to set the number of confirmations desired for which to receive an update. You’ll receive an updated TX for every confirmation up to this amount. The maximum allowed is 10; if not set, it will default to 6.

Another issue it that potentially the websocket never arrives. There's some keep-alive pinging code to maintain the websocket connection, but I'd imagine there are edge cases were it doesn't work. In the page I loaded where it didn't work, the test pings were firing.

**tldr: Are you sure the websocket "endpoint" is working correctly?**

**Comment by acityinohio** [comment]
I think you should be able to just send a hash (and not the hash + address).

But beyond that, the "tx-confirmation" on the WebSockets API is not working properly at the moment (@matthieu is aware of the issue); I think it cuts off after a single confirmation, even if the *confirmation* field has a higher number. Once that's fixed the tx page should work as expected.

**Comment by mflaxman** [comment]
Ok, will close for now as there's nothing to do until the endpoint issue is resolved.
